iml-agent/src/network.rs

```diff
@@ -0,0 +1,449 @@
+// Copyright (c) 2020 DDN. All rights reserved.
+// Use of this source code is governed by a MIT-style
+// license that can be found in the LICENSE file.
+
+use combine::{
+    attempt, choice,
+    error::ParseError,
+    many1, optional,
+    parser::{
+        char::{char, digit, letter, spaces, string},
+        repeat::take_until,
+    },
+    sep_by1,
+    stream::Stream,
+    token, Parser,
+};
+use std::collections::HashMap;
+
+#[derive(Debug, PartialEq, Eq)]
+pub enum InterfaceProperties {
+    InterfaceFlagsAndAttributes((String, Vec<String>, HashMap<String, String>)),
+    InterfaceTypeAndMacAddress((Option<String>, Option<String>)),
+    Inet4AddressAndPrefix((String, u32)),
+    Inet6AddressAndPrefix((String, u32)),
+}
+
+#[derive(Debug)]
+pub struct Address {
```

Comment:
You should probably use https://docs.rs/ipnetwork/ here instead.

Reply:
Makes sense to me. I'll update.

Reply:
Updated.